# Task 1 - Repository cloning and file discovery

The assigned repository is [huggingface/optimum](https://github.com/huggingface/optimum). It is cloned with `--depth 1`; only its URL and locked commit SHA are submitted, not a duplicate of the third-party source tree.

In [1]:
# Executed before the replay branch changed seven lines
!python -m cpg_parser discover --repo ../source-repo

{
  "repo_path": "D:\\Project\\BigData\\Lab04\\source-repo",
  "repo_url": "https://github.com/huggingface/optimum.git",
  "commit_sha": "a6c775e11118d62712057bd3a8c5649898a5312d",
  "raw_python_files": 74,
  "processed_python_files": 61,
  "excluded_python_files": 13,
  "total_lines": 13807,
  "parseable_files": 61,
  "parse_success_rate": 1.0,
  "has_branch": true,
  "has_loop": true,
  "has_call": true,
  "files": [
    "docs/combine_docs.py",
    "docs/conftest.py",
    "optimum/commands/base.py",
    "optimum/commands/env.py",
    "optimum/commands/export/base.py",
    "optimum/commands/optimum_cli.py",
    "optimum/configuration_utils.py",
    "optimum/conftest.py",
    "optimum/exporters/base.py",
    "optimum/exporters/error_utils.py",
    "optimum/exporters/tasks.py",
    "optimum/exporters/utils.py",
    "optimum/fx/__init__.py",
    "optimum/fx/optimization/__init__.py",
    "optimum/fx/optimization/transformations.py",
    "optimum/fx/parallelization/__init__.py",
    "opti

## Reflection

Optimum falls inside the planned selection envelope: 74 raw Python files, 61 processed files, 13,807 lines, and a 100% AST parse rate. The report deliberately separates raw and filtered counts.